src/github/copilotApi.ts

```diff
@@ -0,0 +1,50 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import fetch from 'cross-fetch';
+
+export interface RemoteAgentJobPayload {
+	problem_statement: string;
+	pull_request?: {
+		title?: string;
+		body_placeholder?: string;
+		body_suffix?: string;
+		base_ref?: string;
+	};
+	run_name?: string;
+}
+
+export class CopilotApi {
+	constructor(private token: string) { }
+
+	private get baseUrl(): string {
+		return 'https://api.githubcopilot.com';
+	}
+
+	async postRemoteAgentJob(
+		owner: string,
+		name: string,
+		payload: RemoteAgentJobPayload,
+	): Promise<string> {
+		const repoSlug = `${owner}/${name}`;
+		const apiUrl = `${this.baseUrl}/agents/swe/jobs/${repoSlug}`;
+		const response = await fetch(apiUrl, {
+			method: 'POST',
+			headers: {
+				'Copilot-Integration-Id': 'copilot-developer-dev',
+				'Authorization': `Bearer ${this.token}`,
```

Comment:
If this is using a GitHub auth token, this API may fail in Codespaces Web. Might be worth testing out.

Reply:
This is indeed an issue in Codespaces web.  Thanks for highlighting

Reply:
Resolved this with GitHub